src/libs/actions/Report.js

```diff
@@ -959,22 +950,10 @@ function deleteReportComment(reportID, reportAction) {
         },
     ];
 
-    const report = ReportUtils.getReport(reportID);
-    if (report && report.parentReportActionID) {
-        const parentReportAction = ReportActionsUtils.getParentReportAction(report);
-        if (parentReportAction && parentReportAction.reportActionID) {
-            optimisticData.push({
-                onyxMethod: Onyx.METHOD.MERGE,
-                key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${report.parentReportID}`,
-                value: {
-                    [parentReportAction.reportActionID]: ReportUtils.updateOptimisticParentReportAction(
-                        parentReportAction,
-                        optimisticReport.lastVisibleActionCreated,
-                        CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE,
-                    ),
-                },
-            });
-        }
+    // Update optimistic data for parent report action if the report is a child report
+    const optimisticParentReportData = ReportUtils.getOptimisticDataForParentReportAction(reportID, optimisticReport.lastVisibleActionCreated, CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE);
+    if (!_.isEmpty(optimisticParentReportData)) {
+        optimisticData.push(optimisticParentReportData);
```

Comment:
Coming from https://github.com/Expensify/App/issues/26012:
I won't say it's a regression but it could have been perfect if below case was handled as well:

> If deleted message in thread has visible child actions, `childVisibleActionCount` of `parentReportAction` should not be decreased